Common/src/main/resources/data/crafttweaker/scripts/crafttweaker_sequences_cauldron_saveddata.zs

```diff
@@ -0,0 +1,248 @@
+/*
+    A Sequence is a set of tasks that can be composed and ran.
+
+    With a Sequence, you can run a task, wait a set amount of ticks, and then run another task,
+    without having to worry about the boilerplate that would usually come with doing that.
+
+    Sequences are *currently* not persisted, this means that if a server stops while a sequence is running,
+    then it will **not** be restarted, so while you can make a sequence wait 5 in-game days before another task,
+    if the server restarts while it is waiting for those days, it will not be resumed at that step.
+
+    You can achieve some persistence with tasks by making use of SavedData, an example of that will be below.
+
+    This file will have examples for Cauldron interactions, SavedData and obviously Sequences
+*/
+
+import crafttweaker.api.data.MapData;
+import crafttweaker.api.world.InteractionResult;
+import crafttweaker.api.entity.MobSpawnType;
+import crafttweaker.api.util.sequence.SequenceType;
+import crafttweaker.api.world.Level;
+import crafttweaker.api.world.ServerLevel;
+import crafttweaker.api.data.BoolData;
+
+// The following snippet will run a sequence when an empty cauldron is interacted with while the player is holding a dirt block
+// The sequence that is ran will check if the block has redstone power, if it does not, it will tell the player that it needs
+// redstone, then it will wait until it has redstone power and then wait 20 ticks before spawning fireworks.
+cauldron.addEmptyInteraction(<item:minecraft:dirt>, (blockState, cLevel, bPos, playerIn, hand, stack) => {
+
+    // We only want to run our sequence on the server *thread* (note, a single player world does have a server thread).
+    if !cLevel.isClientSide {
+            // We need to store the parameters that we want to use into local variables before we can use them (this is a bug in the ZenCode engine that will be fixed).
+            var pos = bPos;
+            var block = blockState;
+            var player = playerIn;
+            // Starts a SequenceBuilder
+            cLevel.sequence()
+                // Run immediately
+                .run(level => {
+                    // Check if the redstone signal at the position of the cauldron is 0
+                    if level.getBestNeighborSignal(pos) == 0 {
+                        // Tell the player that it needs redstone
+                        player.sendMessage("Please provide redstone power!");
+                    }
+                })
+                // Sleep until the condition is met
+                .sleepUntil(level => {
+                    // Check if the block has a redstone signal greater than 0
+                    return level.getBestNeighborSignal(pos) > 0;
+                })
+                // Sleep for 20 ticks
+                .sleep(20)
+                // Run immediately ('then' is an alias for 'run', you can use whichever sounds better and is easier to read)
```

Comment:
It might be a good idea to word this a little better. "Run immediately" makes me think this can be used to skip the sequence waiting, while in reality this will be run after the scheduled time has passed/condition has been met.
I wouldn't want users to be confused by the example: maybe something like "Run as soon as the sleep is done"?